invenio_groups/models.py

```diff
@@ -0,0 +1,775 @@
+
+    OPEN = 'O'
+    """Users can self-subscribe."""
+
+    APPROVAL = 'A'
+    """Users can self-subscribe but requires administrator approval."""
+
+    CLOSED = 'C'
+    """Subscription is by administrator invitation only."""
+
+    @classmethod
+    def validate(cls, policy):
+        """Validate subscription policy value."""
+        return policy in [cls.OPEN, cls.APPROVAL, cls.CLOSED]
+
+
+class PrivacyPolicy(object):
+
+    """Group privacy policies."""
+
+    PUBLIC = 'P'
+    """Group membership is fully public."""
+
+    MEMBERS = 'M'
+    """Group administrators and group members can view members."""
+
+    ADMINS = 'A'
+    """Group administrators can view members."""
+
+    @classmethod
+    def validate(cls, policy):
+        """Validate privacy policy value."""
+        return policy in [cls.PUBLIC, cls.MEMBERS, cls.ADMINS]
+
+
+class MembershipState(object):
+
+    """Membership state."""
+
+    PENDING_ADMIN = 'A'
+    """Pending admin verification."""
+
+    PENDING_USER = 'U'
+    """Pending user verification."""
+
+    ACTIVE = 'M'
+    """Active membership."""
+
+    @classmethod
+    def validate(cls, state):
+        """Validate state value."""
+        return state in [cls.ACTIVE, cls.PENDING_ADMIN, cls.PENDING_USER]
+
+
+class Group(db.Model):
+
+    """Group data model."""
+
+    __tablename__ = 'group'
+
+    PRIVACY_POLICIES = [
+        (PrivacyPolicy.PUBLIC, _('Public')),
+        (PrivacyPolicy.MEMBERS, _('Group members')),
+        (PrivacyPolicy.ADMINS, _('Group admins')),
+    ]
+    """Privacy policy choices."""
+
+    SUBSCRIPTION_POLICIES = [
+        (SubscriptionPolicy.OPEN, _('Open')),
+        (SubscriptionPolicy.APPROVAL, _('Open with approval')),
+        (SubscriptionPolicy.CLOSED, _('Closed')),
+    ]
+    """Subscription policy choices."""
+
+    id = db.Column(db.Integer(15, unsigned=True), nullable=False,
+                   primary_key=True, autoincrement=True)
+    """Group identifier."""
+
+    name = db.Column(
+        db.String(255), nullable=False, unique=True, index=True,
+        info=dict(
+            label=_("Name"),
+            description=_('Required. A name of a group.')
+        ))
+    """Name of group."""
+
+    description = db.Column(
+        db.Text, nullable=True, default='',
+        info=dict(
+            label=_("Description"),
+            description=_('Optional. A short description of the group.'
+                          ' Default: Empty'),
+        ))
+    """Description of group."""
+
+    is_managed = db.Column(db.Boolean, default=False, nullable=False)
+    """Determine if group is system managed."""
+
+    privacy_policy = db.Column(
+        ChoiceType(PRIVACY_POLICIES, impl=db.String(1)), nullable=False,
+        default=PrivacyPolicy.ADMINS,
+        info=dict(
+            label=_('Privacy Policy'),
+            description=dict([
+                (PrivacyPolicy.PUBLIC,
+                 _('Group membership is fully public.')),
```

Comment:
I think it's better that these descriptions are added on the ``PrivacyPolicy`` class.